### PR TITLE
Authenticate requests to the /provision endpoint

### PR DESCRIPTION
**Pull request by Half-Shot**

### CODE REVIEW

File: src/provisioning/Provisioner.ts [hunk 1]
```diff
@@ -132,12 +132,18 @@ export class Provisioner {
...
+                    error: "Bad token supplied,"
+                });
+            }
             if (this.isProvisionRequest(req)) {
                 res.header("Access-Control-Allow-Origin", "*");
                 res.header("Access-Control-Allow-Headers",
                     "Origin, X-Requested-With, Content-Type, Accept");
             }
-            next();
+            return next();
```
Comment (jaller94) [inline src/provisioning/Provisioner.ts:146]: What does returning `next()` do? I've never seen that as a pattern.

File: src/provisioning/Provisioner.ts [hunk 2]
```diff
@@ -132,12 +132,18 @@ export class Provisioner {
 
         // Deal with CORS (temporarily for s-web)
         app.use((req, res, next) => {
+            if (!this.ircBridge.getAppServiceBridge().requestCheckToken(req)) {
+                return res.status(403).send({
+                    errcode: "M_FORBIDDEN",
+                    error: "Bad token supplied,"
```
Comment (jaller94) [inline src/provisioning/Provisioner.ts:138]:
```suggestion
                    error: "Bad token supplied"
```

File: src/provisioning/Provisioner.ts [hunk 3]
```diff
@@ -132,12 +132,18 @@ export class Provisioner {
 
         // Deal with CORS (temporarily for s-web)
```
Comment (jaller94) [inline src/provisioning/Provisioner.ts:133]: This comment which once described the entire purpose of this handler has now become outdated.

**Comment by Half-Shot** [comment]
>When there are many projects already using matrix-appservice-bridge, why don't we release a types/matrix-appservice-bridge package instead of keeping the types updated locally?

We should probably just bundle a definition with the library itself. I've not done it yet due to time

**Comment by Cogitri** [comment]
Hey,

since this is marked as `SECURITY FIX` in the release notes, should this get a CVE? Does this also apply to older releases? Without a CVE (or similiar tracking information) distros don't really know how to act upon this.

**Comment by Half-Shot** [comment]
We're not treating it as high enough severity to warrant a CVE (which is why it didn't get a minor release and has been tacked onto the next release). It only affects users who enabled provisioning and didn't firewall the application out from the rest of the internet. We've determined this to be of low severity because the provisioning system is useless unless you are also running an integration manager (and the feature is off by default), and you'd have to be exposing your appservice to the world.

**Comment by Cogitri** [comment]
Hm, I personally I feel like even for low severity security issues a CVE should be issued, otherwise the chance for people who happen to run the appservice in that configuration (for whatever reason), won't know about that security issue.

**Pull request merged**